pretranslated/14/index.d

```diff
@@ -69,7 +69,16 @@ alias CXIndex = void*;
 /**
  * \brief A single translation unit, which resides in an index.
  */
-struct CXTranslationUnitImpl;
+struct CXTranslationUnitImpl {
```

Comment:
clang from 14 to at least 19 uses same struct

Reply:
Why do we need the definition of it though?

Reply:
To hide a pointer to the `TranslationUnit`  inside of one of the unused fields (`CommentToXML`), so that this pointer is accessible to any cursor

Reply:
Other ways involve massive Cursor interface changes

Reply:
We would change the interface of a type that's from libclang??

Reply:
There are no changes here, this is just same declaration as in clang header: 

https://github.com/llvm/llvm-project/blob/e5865ec93cbf77f84c55e55383ce3254186fc065/clang/tools/libclang/CXTranslationUnit.h#L28
```hpp
struct CXTranslationUnitImpl {
  clang::CIndexer *CIdx;
  clang::ASTUnit *TheASTUnit;
  clang::cxstring::CXStringPool *StringPool;
  void *Diagnostics;
  void *OverridenCursorsPool;
  clang::index::CommentToXMLConverter *CommentToXML;
  unsigned ParsingOptions;
  std::vector<std::string> Arguments;
};
```

Reply:
I don't think depending on the internals of a libclang struct is a good idea.

Reply:
It solves leak problem at least for the next few years with little effort

Before that I also tried traditional way of adding a pointer inside the D cursor structure - it causes big changes in all interfaces inside of dpp because it decoreated by all these `@safe @nogc pure const`

Reply:
"with little effort" - I disagree that this is little effort.

> it causes big changes in all interfaces inside of dpp because it decoreated by all these @safe @nogc pure const

This is interesting. Why is that?

Reply:
> Why is that?

As I remember, `Cursor` becomes mutable and it is leads to changes in almost all functions what use `Cursor`

Reply:
It would be counter productive to mark code with opaque memory leaks as safe. 
Regarding the supported version, that can be easily resolved by bumping the major version. There's no point in supporting old liblang libraries.

Reply:
> It would be counter productive to mark code with opaque memory leaks as safe.

This is a reason for another PR, there is no need to clutter this PR

Reply:
I meant that it's better to rely on the definition to fix the memory leaks instead of trying to keep the library compatible with older versions of libclang.

Reply:
>  instead of trying to keep the library compatible with older versions of libclang.

I don't have such a goal - I'm only worried about the interface changes that affect `dpp` (it heavy uses `libclang`)